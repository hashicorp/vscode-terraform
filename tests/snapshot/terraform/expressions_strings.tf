"hello"

"\n\r\t\"\\"

"$${}"

"%%{}"

thing = <<EOT
hello
world
EOT

block {
  value = <<EOT
hello
world
EOT
}

block {
  value = <<-EOT
  hello
    world
  EOT
}

"Hello, ${var.name}!"

"Hello, %{ if var.name != "" }${var.name}%{ else }unnamed%{ endif }!"

<<EOT
%{ for ip in aws_instance.example.*.private_ip }
server ${ip}
%{ endfor }
EOT

<<EOT
%{ for ip in aws_instance.example.*.private_ip ~}
server ${ip}
%{ endfor ~}
EOT


