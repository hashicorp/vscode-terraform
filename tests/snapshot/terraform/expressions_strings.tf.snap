>"hello"
#^ source.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
# ^^^^^ source.terraform string.quoted.double.terraform
#      ^ source.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>
>"\n\r\t\"\\"
#^ source.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
# ^^ source.terraform string.quoted.double.terraform constant.character.escape.terraform
#   ^^ source.terraform string.quoted.double.terraform constant.character.escape.terraform
#     ^^ source.terraform string.quoted.double.terraform constant.character.escape.terraform
#       ^^ source.terraform string.quoted.double.terraform constant.character.escape.terraform
#         ^^ source.terraform string.quoted.double.terraform constant.character.escape.terraform
#           ^ source.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>
>"$${}"
#^ source.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
# ^^^^ source.terraform string.quoted.double.terraform
#     ^ source.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>
>"%%{}"
#^ source.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
# ^^^^ source.terraform string.quoted.double.terraform
#     ^ source.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>
>thing = <<EOT
#^^^^^ source.terraform variable.declaration.terraform variable.other.readwrite.terraform
#     ^ source.terraform variable.declaration.terraform
#      ^ source.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#       ^ source.terraform variable.declaration.terraform
#        ^^ source.terraform string.unquoted.heredoc.terraform keyword.operator.heredoc.terraform
#          ^^^ source.terraform string.unquoted.heredoc.terraform keyword.control.heredoc.terraform
>hello
#^^^^^^ source.terraform string.unquoted.heredoc.terraform
>world
#^^^^^^ source.terraform string.unquoted.heredoc.terraform
>EOT
#^^^^ source.terraform string.unquoted.heredoc.terraform keyword.control.heredoc.terraform
>
>block {
#^^^^^ source.terraform meta.block.terraform entity.name.label.terraform
#     ^ source.terraform meta.block.terraform
#      ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  value = <<EOT
#^^ source.terraform meta.block.terraform
#  ^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#       ^ source.terraform meta.block.terraform variable.declaration.terraform
#        ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#         ^ source.terraform meta.block.terraform variable.declaration.terraform
#          ^^ source.terraform meta.block.terraform string.unquoted.heredoc.terraform keyword.operator.heredoc.terraform
#            ^^^ source.terraform meta.block.terraform string.unquoted.heredoc.terraform keyword.control.heredoc.terraform
>hello
#^^^^^^ source.terraform meta.block.terraform string.unquoted.heredoc.terraform
>world
#^^^^^^ source.terraform meta.block.terraform string.unquoted.heredoc.terraform
>EOT
#^^^^ source.terraform meta.block.terraform string.unquoted.heredoc.terraform keyword.control.heredoc.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>block {
#^^^^^ source.terraform meta.block.terraform entity.name.label.terraform
#     ^ source.terraform meta.block.terraform
#      ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  value = <<-EOT
#^^ source.terraform meta.block.terraform
#  ^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#       ^ source.terraform meta.block.terraform variable.declaration.terraform
#        ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#         ^ source.terraform meta.block.terraform variable.declaration.terraform
#          ^^^ source.terraform meta.block.terraform string.unquoted.heredoc.terraform keyword.operator.heredoc.terraform
#             ^^^ source.terraform meta.block.terraform string.unquoted.heredoc.terraform keyword.control.heredoc.terraform
>  hello
#^^^^^^^^ source.terraform meta.block.terraform string.unquoted.heredoc.terraform
>    world
#^^^^^^^^^^ source.terraform meta.block.terraform string.unquoted.heredoc.terraform
>  EOT
#^^^^^^ source.terraform meta.block.terraform string.unquoted.heredoc.terraform keyword.control.heredoc.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>"Hello, ${var.name}!"
#^ source.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
# ^^^^^^ source.terraform string.quoted.double.terraform
#       ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#        ^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#          ^^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform support.constant.terraform
#             ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.operator.accessor.terraform
#              ^^^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform variable.other.member.terraform
#                  ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
#                   ^ source.terraform string.quoted.double.terraform
#                    ^ source.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>
>"Hello, %{ if var.name != "" }${var.name}%{ else }unnamed%{ endif }!"
#^ source.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
# ^^^^^^ source.terraform string.quoted.double.terraform
#       ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#        ^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#          ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#           ^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.control.terraform
#             ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#              ^^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform support.constant.terraform
#                 ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.operator.accessor.terraform
#                  ^^^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform variable.other.member.terraform
#                      ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#                       ^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.operator.terraform
#                         ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#                          ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                           ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
#                            ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#                             ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
#                              ^^^^^^^^^^ source.terraform string.quoted.double.terraform
#                                        ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#                                         ^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#                                           ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#                                            ^^^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.control.terraform
#                                                ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#                                                 ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
#                                                  ^^^^^^ source.terraform string.quoted.double.terraform
#                                                        ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#                                                         ^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#                                                           ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#                                                            ^^^^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.control.terraform
#                                                                 ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#                                                                  ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
#                                                                   ^ source.terraform string.quoted.double.terraform
#                                                                    ^ source.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>
><<EOT
#^^ source.terraform string.unquoted.heredoc.terraform keyword.operator.heredoc.terraform
#  ^^^ source.terraform string.unquoted.heredoc.terraform keyword.control.heredoc.terraform
>%{ for ip in aws_instance.example.*.private_ip }
#^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#  ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#   ^^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.control.terraform
#      ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#       ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform variable.other.readwrite.terraform
#         ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#          ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.control.terraform
#            ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#             ^^^^^^^^^^^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform variable.other.readwrite.terraform
#                         ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.operator.accessor.terraform
#                          ^^^^^^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform variable.other.member.terraform
#                                 ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.operator.accessor.terraform
#                                  ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#                                    ^^^^^^^^^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform variable.other.member.terraform
#                                              ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#                                               ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
>server ${ip}
#^^^^^^ source.terraform string.unquoted.heredoc.terraform
#      ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#       ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#         ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform variable.other.readwrite.terraform
#           ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
>%{ endfor }
#^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#  ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#   ^^^^^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.control.terraform
#         ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#          ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
>EOT
#^^^^ source.terraform string.unquoted.heredoc.terraform keyword.control.heredoc.terraform
>
><<EOT
#^^ source.terraform string.unquoted.heredoc.terraform keyword.operator.heredoc.terraform
#  ^^^ source.terraform string.unquoted.heredoc.terraform keyword.control.heredoc.terraform
>%{ for ip in aws_instance.example.*.private_ip ~}
#^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#  ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#   ^^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.control.terraform
#      ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#       ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform variable.other.readwrite.terraform
#         ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#          ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.control.terraform
#            ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#             ^^^^^^^^^^^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform variable.other.readwrite.terraform
#                         ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.operator.accessor.terraform
#                          ^^^^^^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform variable.other.member.terraform
#                                 ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.operator.accessor.terraform
#                                  ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#                                    ^^^^^^^^^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform variable.other.member.terraform
#                                              ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.operator.template.right.trim.terraform
#                                                ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
>server ${ip}
#^^^^^^ source.terraform string.unquoted.heredoc.terraform
#      ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#       ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#         ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform variable.other.readwrite.terraform
#           ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
>%{ endfor ~}
#^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#  ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform
#   ^^^^^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.control.terraform
#         ^^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.operator.template.right.trim.terraform
#           ^ source.terraform string.unquoted.heredoc.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
>EOT
#^^^^ source.terraform string.unquoted.heredoc.terraform keyword.control.heredoc.terraform
>
>
>